### PR TITLE
chore: Create new docs site

### PR DESCRIPTION
**Pull request by mayank99**
The code lives in a new workspace called `website`. To run locally, pull the branch, run `yarn` to install deps, then run `yarn dev` or `yarn dev --filter=website`.

Pending tasks (can come in separate PRs):

- real content (currently most of it is copy pasted from UX website)
- currently deployed to netlify (needs logo in the footer if we keep it)
- search / algolia
- Light theme
- SEO with proper canonical url (tied with domain name)
- Icons and layouts to be integrated into this same site.

TBD:
- add some tests
- change deployment from netlify to azure

---

Deploy prevew: https://mayank-website--itwinui.netlify.app/

### CODE REVIEW

File: apps/website/src/pages/index.astro
Comment (mayank99) [inline apps/website/src/pages/index.astro:186]: @gretanausedaite What are the changes in this commit ([`b2158c1`](https://github.com/iTwin/iTwinUI-react/pull/785/commits/b2158c15ac611e3d548c29bad3dc24c8ed3f4043)) for? We should not really be using flex order or reverse direction (it's confusing for users as well as for our own code maintenance). I thought our design was already responsive.

Comment (gretanausedaite) [reply apps/website/src/pages/index.astro:186]: I updated sizes to fit mobile screens better - I feel like margins were too big before.
I use flex order to arrange hero box items as our designer wanted. Previously we were hiding logo in mobile version.
Removed reverse-direction for consistency.

**Comment by rohan-kadkol** [comment]
@mayank99 Since you and Greta worked on this, do you maybe want to assign another reviewer from group 1? Because I think Greta got automatically assigned.

**Comment by gretanausedaite** [comment]
> @mayank99 Since you and Greta worked on this, do you maybe want to assign another reviewer from group 1? Because I think Greta got automatically assigned.

We can do smaller scale reviews for this one as it is not user facing code (users will not use this code). 
Reviewers could check website `components` folder and run the app, click through it and see if something is working not as expected.

**Review by r100-stack — COMMENTED** [review]
* If this is a small task, might be good to be able to switch the theme inside the rectangle. That way users can see the colors in the light and dark theme.
  * image
* I'm not sure if this is the correct link, but this section doesn't appear correctly for me
  * image
* When loading a code sandbox example, one icon goes out of bounds. Also there is a scrollbar. The icon out of bounds also happens on the [sandpack](https://sandpack.codesandbox.io/) website, but they don't have a scrollbar. Maybe we can remove the scrollbar.
##### Our website (Alert -> Informational)
image
##### Sandpack website
image
* Not sure if this is just me, but clicking a new component from the left side menu takes slightly longer to load the new page. Long loading times might be hard to see multiple components one after the other.
* The hashtag icon for the section might be a bit confusing to some users. Since usually the titles are not clickable. Only the # is clickable (E.g. [Supabase docs](https://supabase.com/docs/reference/javascript/)). Also, the hashtag disappears when trying to click it.
##### Our docs
[vokoscreenNG-2022-10-28_12-21-24.webm](https://user-images.githubusercontent.com/45748283/198686151-c9007a38-5216-4db7-b408-87fcc27c30ea.webm)
##### Supabase docs
[vokoscreenNG-2022-10-28_12-23-42.webm](https://user-images.githubusercontent.com/45748283/198686261-9f67e40a-8e52-453d-883f-0a7a64fbb446.webm)
* Might be helpful to have a "copy all" text button for the code
image
* "Note" as text might be easy to miss. An iTwinUI alert (type: warning/information) component might be better. E.g. Getting started page
image
* Avatar -> Related resources seems empty
image
* Breadcrumbs page has `!!...!!` in plain text. Maybe that was intended to be formatted some other way
image
* Carousel page has incomplete fields
* Not sure if this is helpful for users, but maybe can add a question mark to specify which props are optional.
* Datepicker page's default column can wrap the text instead of adding a horizontal scroll on the table
image
* Dropdown menu gives an error 
image
* Error page's code sandbox's buttons don't seem to be fully clickable. The top half is clickable but the bottom half is not. Maybe the sandpack is adding a transparent layer at the bottom for the two left circle buttons and the view code right button. That might be interfering with the inner content
[vokoscreenNG-2022-10-28_12-48-53.webm](https://user-images.githubusercontent.com/45748283/198690729-914528aa-4051-46b0-ac57-480a4d833b4e.webm)
image
* ErrorPage's errorMessage description's code block overlaps over each other from multiple lines
image
* InformationPanel's props have default in the description
image
  * Other similar places: `Checkbox`, `LabeledInput -> displayStyle`, `LabeledSelect`, `Radio & Radio Tile -> setFocus`, `TextArea -> setFocus`, `ToggleSwitch`, `Typography`, `Anchor`
* SideNavigation gives an error
image
* Surface props are missing
* Tile page code sandbox has errors
image
* Tree code sandbox has errors
image
* Not sure, but are the datatypes of `Button->as` correct? Same for `ButtonGroup`, other buttons, and also `Text`
image
* IconButton is giving error
image
* Checkbox props default is in the description
image
* InputGroup has errors
image
* `Select -> popoverProps` has @link tag
* `ToggleSwitch` code sandbox has errors
image

**Comment by mayank99** [comment]
@rohan-kadkol  I've fixed the missing styles and sandpack errors. The UI bugs related to sandpack don't need to be addressed now. Same with props table. And the rest are missing features or content related, which we can address in separate PRs as well.

Navigating between pages is expected to be a bit slow locally because it's generating the page each time, but it should be fast on the deployed version: https://mayank-website--itwinui.netlify.app/

**Review by r100-stack — APPROVED** [review]

**Comment by rohan-kadkol** [comment]
> @rohan-kadkol I've fixed the missing styles and sandpack errors. The UI bugs related to sandpack don't need to be addressed now. Same with props table. And the rest are missing features or content related, which we can address in separate PRs as well.

@mayank99 I approved the PR to avoid holding back the release, since like you said, we can do the improvements in future PRs. But when testing your latest two commits, I get this error. Don't know if it is a problem on my end or because of the latest changes.

image

**Review by gretanausedaite — APPROVED** [review]
Should check if all React and Storybook tests pass as we changed some components.

**Comment by mayank99** [comment]
> when testing your latest two commits, I get this error. Don't know if it is a problem on my end or because of the latest changes.
> 
> image

This should be fixed now.

**Comment by mayank99** [comment]
I was getting ready to merge this when I realized that #845 would make this fail because we are relying on internal imports for generating props table. Need to investigate.

**Comment by mayank99** [comment]
Just like I feared, I am getting this error:
```
Rollup failed to resolve import "@itwin/itwinui-react/esm/core/Dialog" from "src/examples/Dialog.static.tsx"
```

**Comment by mayank99** [comment]
> Just like I feared, I am getting this error:
> 
> ```
> Package subpath './src/core/InformationPanel/InformationPanel.tsx' is not defined by "exports" in /node_modules/@itwin/itwinui-react/package.json
> ```

Fixed in [`c665ae2`](https://github.com/iTwin/iTwinUI-react/pull/785/commits/c665ae2a63775f03532b649ae72a22209e7eab13) (tested with and without export maps from #845).

**Comment by mayank99** [comment]
Tests are passing so I'm merging this. yolo

**Pull request merged**
